tests/rules/test_float_values.py

```diff
@@ -0,0 +1,141 @@
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from tests.common import RuleTestCase
+
+
+class FloatValuesTestCase(RuleTestCase):
+    rule_id = 'float-values'
+
+    def test_disabled(self):
+        conf = (
+            'float-values: disable\n'
+            'new-line-at-end-of-file: disable\n'
+            'document-start: disable\n'
+        )
+        self.check('angle: 0.0', conf)
+        self.check('angle: .NaN', conf)
+        self.check('angle: .INF', conf)
+        self.check('angle: .1', conf)
+        self.check('angle: 10e-6', conf)
+
+    def test_numeral_before_decimal(self):
+        conf = (
+            'float-values:\n'
+            '  require-numeral-before-decimal: true\n'
+            '  forbid-scientific-notation: false\n'
+            '  forbid-nan: false\n'
+            '  forbid-inf: false\n'
+            'new-line-at-end-of-file: disable\n'
+            'document-start: disable\n'
+        )
+        self.check('angle: .1', conf, problem=(1, 8))
+        self.check('angle: 0.0', conf)
+        self.check('angle: \'0.0\'', conf)
+        self.check('angle: !custom_tag 0.0', conf)
+        self.check(
+            '- &angle 0.0\n'
+            '- *angle\n',
+            conf
+        )
+        self.check(
+            '- &angle .0\n'
+            '- *angle\n',
+            conf,
+            problem=(1, 10)
+        )
+
+    def test_scientific_notation(self):
+        conf = (
+            'float-values:\n'
+            '  require-numeral-before-decimal: false\n'
+            '  forbid-scientific-notation: true\n'
+            '  forbid-nan: false\n'
+            '  forbid-inf: false\n'
+            'new-line-at-end-of-file: disable\n'
+            'document-start: disable\n'
+        )
+        self.check('angle: 10e6', conf, problem=(1, 8))
+        self.check('angle: 10e-6', conf, problem=(1, 8))
+        self.check('angle: 0.00001', conf)
+        self.check('angle: \'10e-6\'', conf)
+        self.check('angle: !custom_tag 10e-6', conf)
+        self.check(
+            '- &angle 0.000001\n'
+            '- *angle\n',
+            conf
+        )
+        self.check(
+            '- &angle 10e-6\n'
+            '- *angle\n',
+            conf,
+            problem=(1, 10)
+        )
+        self.check(
+            '- &angle 10e6\n'
+            '- *angle\n',
+            conf,
+            problem=(1, 10)
+        )
+
+    def test_nan(self):
+        conf = (
+            'float-values:\n'
+            '  require-numeral-before-decimal: false\n'
+            '  forbid-scientific-notation: false\n'
+            '  forbid-nan: true\n'
+            '  forbid-inf: false\n'
+            'new-line-at-end-of-file: disable\n'
+            'document-start: disable\n'
+        )
+        self.check('angle: .NaN', conf, problem=(1, 8))
+        self.check('angle: .NAN', conf, problem=(1, 8))
+        self.check('angle: \'NaN\'', conf)
```

Comment:
Do you mean 
```suggestion
        self.check('angle: \'.NaN\'', conf)
```
?
To make sure `.NaN` is not reported if it's a string?

Otherwise, I don't understand why this line is for, can you explain?

Reply:
Same with this one, I am guessing this was my intent.